serde_with_macros/src/lib.rs

```diff
@@ -436,6 +487,243 @@ fn field_has_attribute(field: &Field, namespace: &str, name: &str) -> bool {
     false
 }
 
+/// Similar to `skip_serializing_none`, but takes an `argument` which is then used with
+/// `serde(skip_serializing_if = "argument")`
+///
+/// The attribute can be added to structs and enums.
+/// The `#[skip_serializing_if]` attribute must be placed *before* the `#[derive]` attribute.
+///
+/// # Example
+///
+/// Let's say that we have an untagged enum, and we want to skip serialization for the variant that
+/// is `empty`
+///
+/// ```rust
+/// use serde::Serialize;
+///
+/// #[derive(Serialize)]
+/// #[serde(untagged)]
+/// enum Wrapper {
+///     Nothing,
+///     String(String),
+///     Int32(i32),
+/// }
+///
+/// impl Wrapper {
+///     fn is_nothing(&self) -> bool {
+///         match self {
+///             Self::Nothing => true,
+///             _ => false,
+///         }
+///     }
+/// }
+///
+/// #[derive(Serialize)]
+/// struct Data {
+///     #[serde(skip_serializing_if = "Wrapper::is_nothing")]
+///     a: Wrapper,
+///     #[serde(skip_serializing_if = "Wrapper::is_nothing")]
+///     b: Wrapper,
+///     #[serde(skip_serializing_if = "Wrapper::is_nothing")]
+///     c: Wrapper,
+///     #[serde(skip_serializing_if = "Wrapper::is_nothing")]
+///     d: Wrapper,
+/// }
+/// ```
+///
+/// The `skip_serializing_if` annotation is repetitive and harms readability.
+/// Instead, the same struct can be written as:
+///
+/// ```rust
+/// use serde::Serialize;
+/// use serde_with_macros::skip_serializing_if;
+///
+/// #[derive(Serialize)]
+/// #[serde(untagged)]
+/// enum Wrapper {
+///     Nothing,
+///     String(String),
+///     Int32(i32),
+/// }
+///
+/// impl Wrapper {
+///     pub fn is_nothing(&self) -> bool {
+///         match self {
+///             Self::Nothing => true,
+///             _ => false,
+///         }
+///     }
+/// }
+///
+/// #[skip_serializing_if("Wrapper::is_nothing")]
```

Comment:
It is no longer necessary for attributes to rely so much on strings. For the most part, any tokens are supported. I think removing the string quotes here would make it nicer. There is also no pre-existing serde interface similar. So confusion or consistency with that does not apply.